cocos/asset/assets/material.ts

```diff
@@ -29,9 +29,10 @@ import { Texture, Type } from '../../gfx';
 import { TextureBase } from './texture-base';
 import { IPassInfoFull, Pass, PassOverrides } from '../../render-scene/core/pass';
 import { MacroRecord, MaterialProperty } from '../../render-scene/core/pass-utils';
-import { Color, warnID, Vec4, cclegacy } from '../../core';
+import { Color, warnID, Vec4, cclegacy, warn } from '../../core';
 import { SRGBToLinear } from '../../rendering/pipeline-funcs';
 import { Renderer } from '../../misc/renderer';
+import type { Root } from '../../root';
```

Comment:
Is it right that `import type` will not introduce dependency?

Reply:
The line of `import type` will omitted after ts compilation. So it will not change the module require order. Just for type checking.

Reply:
I see.